app/gui2/src/stores/graph/index.ts

```diff
@@ -224,6 +218,20 @@ export const useGraphStore = defineStore('graph', () => {
     commitEdit(edit, new Map([[rhs.exprId, meta]]))
   }
 
+  function addMissingImports(edit: MutableModule, newImports: RequiredImport[]) {
+    if (!newImports.length) return
+    const root = moduleRoot.value
+    if (!root) {
+      console.error(`BUG: Cannot add required imports: No module root.`)
+      return
+    }
+    const importsToAdd = filterOutRedundantImports(imports.value, newImports)
+    if (!importsToAdd.length) return
+    // The top level of the module is always a block.
+    const topLevel = astModule.get(root)! as Ast.BodyBlock
```

Comment:
This should be operating on `edit` instead of `astModule`, as it might already have import list modified. Therefore we also can't really use `imports` ref value in this function. Instead, current import list should be recomputed from the edited module.

This should not fail with current logic though, so it's not a merge blocker. But needs to be fixed later.